### PR TITLE
[libc++][test] XFAIL sized deallocation tests for AIX, z/OS, and MinGW

### PR DESCRIPTION
**Pull request by xingxue-ibm**
The sized deallocation test cases fail on AIX, z/OS, and MinGW because they default to `-fno-sized-deallocation`. This patch XFAILs these test cases for the affected targets. Once they change the default, we will get `unexpectedly pass`.

### CODE REVIEW

**Review by daltenty — APPROVED** [review]
LGTM, thanks

**Review by wangpc-pp — APPROVED** [review]
LGTM. Thanks!
(Do we have buildbots for these configurations?)

**Review by ldionne — COMMENTED** [review]
LGTM assuming CI passes. I'd be surprised if this passes on MinGW though since we were already testing this on MinGW and it didn't fail? I don't understand why. Either way, I'm happy if the CI is green, since this matches my expectation of how things should look.

**Review by ldionne — APPROVED** [review]

**Comment by mstorsjo** [comment]
> LGTM. Thanks! (Do we have buildbots for these configurations?)

We don’t have full Buildbot for mingw - but we do have it in the precommit CI using an existing release of Clang.



> LGTM assuming CI passes. I'd be surprised if this passes on MinGW though since we were already testing this on MinGW and it didn't fail? I don't understand why. 

The test is `UNSUPPORTED: clang-18`, which covers the setup in CI - the issue only appears if testing with a nightly snapshot.

File: libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp [hunk 1]
```diff
@@ -15,6 +15,9 @@
 // XFAIL: apple-clang
 // XFAIL: using-built-library-before-llvm-11
 
+// AIX, z/OS, and MinGW default to -fno-sized-deallocation.
+// XFAIL: target={{.+}}-aix{{.*}}, target={{.+}}-zos{{.*}}, target={{.+}}-mingw32{{.*}}
```
Comment (mstorsjo) [inline libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp:19]: The patterns for mingw here should be `{{.*}}-windows-gnu` - the triples get normalized into that form. Please grep for that and see which exact spelling we use elsewhere (can’t easily check myself ATM).

**Review by mstorsjo — CHANGES_REQUESTED** [review]

File: libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp (hunk 1 above)
Comment (mstorsjo) [reply libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp:19]: (Or `{{.+}}` for the arch obviously is fine too.)

Comment (xingxue-ibm) [reply libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp:19]: You are right. Targets used for MinGW are as follows. I will change to use `{{.+}}-windows-gnu` as suggested. Thanks, @mstorsjo!
```
target=x86_64-w64-windows-gnu
target=i686-w64-windows-gnu
```

**Review by mstorsjo — APPROVED** [review]
LGTM, thanks

**Comment by xingxue-ibm** [comment]
> > LGTM. Thanks! (Do we have buildbots for these configurations?)
> 
> We don’t have full Buildbot for mingw - but we do have it in the precommit CI using an existing release of Clang.
> 
> > LGTM assuming CI passes. I'd be surprised if this passes on MinGW though since we were already testing this on MinGW and it didn't fail? I don't understand why.
> 
> The test is `UNSUPPORTED: clang-18`, which covers the setup in CI - the issue only appears if testing with a nightly snapshot.

The AIX precommit CI uses an existing Clang release as well. I don't think there is precommit CI for z/OS.

**Review by mordante — APPROVED** [review]

**Review by zibi2 — APPROVED** [review]

**Pull request merged**